provider/vsphere/config.go

```diff
@@ -99,6 +103,14 @@ func (c *environConfig) enableDiskUUID() bool {
 	return c.attrs[cfgEnableDiskUUID].(bool)
 }
 
+func (c *environConfig) diskProvisioningType() vsphereclient.DiskProvisioningType {
+	provType, ok := c.attrs[cfgDiskProvisioningType]
+	if !ok {
+		return vsphereclient.DiskTypeThin
```

Comment:
Why not return the default,  DiskTypeThickEagerZero, here?

Or add a comment as to why not.

Reply:
Fixed. Thanks!

Added a few more tests that hopefully covers this. Please take another look.